juju/application.py

```diff
@@ -658,20 +658,28 @@ async def set_constraints(self, constraints):
         return await app_facade.SetConstraints(application=self.name, constraints=constraints)
 
     async def refresh(
-            self, channel=None, force=False, force_series=False, force_units=False,
-            path=None, resources=None, revision=None, switch=None):
+            self,
+            channel: Optional[str] = None,
+            force: bool = False,
+            force_series: bool = False,
+            force_units: bool = False,
+            path: Optional[Union[Path, str]] = None,
+            resources: Optional[Dict[str, str]] = None,
+            revision: Optional[int] = None,
+            switch: Optional[str] = None,
+    ):
         """Refresh the charm for this application.
 
-        :param str channel: Channel to use when getting the charm from the
+        :param str|None channel: Channel to use when getting the charm from the
```

Comment:
Do we need the types listed here, now that actual type annotations are present? Are these docstrings actually used to generate reference docs? If so, will it / can it use the type annotations?

Reply:
I don't actually know. I was thinking to try and see.